CHANGELOG.md

```diff
@@ -3,6 +3,203 @@ This project adheres to [Semantic Versioning](https://semver.org/).
+- `ClockId::set_time()` and `time::clock_settime()` are now enabled on macOS
+  ([#2241](https://github.com/nix-rust/nix/pull/2241))
+- Added `IpBindAddressNoPort` sockopt to support `IP_BIND_ADDRESS_NO_PORT`
+  available on linux. ([#2244](https://github.com/nix-rust/nix/pull/2244))
+- Enable `MapFlags::map_hugetlb_with_size_log2` method for Android/Fuchsia
+  ([#2245](https://github.com/nix-rust/nix/pull/2245))
+- Added `TcpFastOpenConnect` sockopt to support `TCP_FASTOPEN_CONNECT`
+  available on linux. ([#2247](https://github.com/nix-rust/nix/pull/2247))
+- Add `reboot(2)` for OpenBSD/NetBSD
+  ([#2251](https://github.com/nix-rust/nix/pull/2251))
+- Added new `MemFdCreateFlag` constants to `sys::memfd` on Linux and Android
+  related to hugetlbfs support.
+  ([#2252](https://github.com/nix-rust/nix/pull/2252))
+- Expose the inner fd of `Kqueue` through:
+
+  * impl AsFd for Kqueue
+  * impl From\<Kqueue\> for OwnedFd
+
+  ([#2258](https://github.com/nix-rust/nix/pull/2258))
+- Added `sys::eventfd` support on FreeBSD
+  ([#2259](https://github.com/nix-rust/nix/pull/2259))
+- Added `MmapFlags::MAP_FIXED` constant in `sys::mman` for netbsd and openbsd
+  ([#2260](https://github.com/nix-rust/nix/pull/2260))
+- Added the `SO_LISTENQLIMIT` sockopt.
+  ([#2263](https://github.com/nix-rust/nix/pull/2263))
+- Enable the `AT_EMPTY_PATH` flag for the `fchownat()` function
+  ([#2267](https://github.com/nix-rust/nix/pull/2267))
+- Add `AtFlags::AT_EMPTY_PATH` for FreeBSD and Hurd
+  ([#2270](https://github.com/nix-rust/nix/pull/2270))
+- Enable `OFlag::O_DIRECTORY for Solarish
+  ([#2275](https://github.com/nix-rust/nix/pull/2275))
+- Added the `Backlog` wrapper type for the `listen` call.
+  ([#2276](https://github.com/nix-rust/nix/pull/2276))
+- Add `clock_nanosleep()` ([#2277](https://github.com/nix-rust/nix/pull/2277))
+- Enabled `O_DIRECT` in `fcntl::OFlags` for solarish
+  ([#2278](https://github.com/nix-rust/nix/pull/2278))
+- Added a new API sigsuspend.
+  ([#2279](https://github.com/nix-rust/nix/pull/2279))
+- - Added `errno::Errno::set` function
+  - Added `errno::Errno::set_raw` function
+  - Added `errno::Errno::last_raw` function
+  - Added `errno::Errno::from_raw` function
+
+  ([#2283](https://github.com/nix-rust/nix/pull/2283))
+- Enable the `AT_EMPTY_PATH` flag for the `linkat()` function
+  ([#2284](https://github.com/nix-rust/nix/pull/2284))
+- Enable unistd::{sync, syncfs} for Android
+  ([#2296](https://github.com/nix-rust/nix/pull/2296))
+
+### Changed
+
+- `poll` now takes `PollTimeout` replacing `libc::c_int`.
+  ([#1876](https://github.com/nix-rust/nix/pull/1876))
+- Deprecated `sys::eventfd::eventfd`.
+  ([#1945](https://github.com/nix-rust/nix/pull/1945))
+- `mmap`, `mmap_anonymous`, `munmap`, `mremap`, `madvise`, `msync`, `mprotect`,
+  `munlock` and `mlock` updated to use `NonNull`.
+  ([#2000](https://github.com/nix-rust/nix/pull/2000))
+- Simplified the function signatures of `recvmmsg` and `sendmmsg`
```

Comment:
Please remember to remove this change before releasing a new version, since this would be a regression of safety (#2120 fixed a safety bug but embarrassingly also introduced a new one, as explained in #2162, revert in #2227).

I'll revisit #2227 later this day and probably make some slight changes to better match the previous version regarding some elided lifetimes (my version should be equivalent, but I should have opened a separate PR to remove the additional lifetimes that became redundant in recent compiler versions instead of doing both things in one PR).

Reply:
> Please remember to remove this change before releasing a new version

Will do it when updating this PR:)

Reply:
Done